src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,72 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
-    private static final Injector injector = new Injector();
+    private static final Injector INJECTOR = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>(
+            Map.of(
+                    FileReaderServiceImpl.class, new FileReaderServiceImpl(),
+                    ProductParserImpl.class, new ProductParserImpl(),
+                    ProductServiceImpl.class, new ProductServiceImpl()
+            )
+    );
 
     public static Injector getInjector() {
-        return injector;
+        return INJECTOR;
+    }
+
+    public <T> T getInstance(Class<T> interfaceClass) {
+        if (instances.containsKey(interfaceClass)) {
+            return interfaceClass.cast(instances.get(interfaceClass));
+        }
+
+        for (Class<?> clazz : instances.keySet()) {
+            if (interfaceClass.isAssignableFrom(clazz)
+                    && clazz.isAnnotationPresent(Component.class)) {
+                try {
+                    T instance = (T) createInstance(clazz);
+                    instances.put(interfaceClass, instance);
+                    return instance;
+                } catch (ExceptionInInitializerError e) {
+                    throw new RuntimeException("Can't create object of the class", e);
+                }
+            }
+        }
+        throw new RuntimeException("No implementation found for " + interfaceClass.getName());
     }
 
-    public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+    private <T> T createInstance(Class<T> clazz) {
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Class " + clazz.getName()
+                    + " is not annotated with @Component");
+        }
```

Comment:
I dont think you'll ever enter this if, because of 31 line. I suggest you check if it's component before 29 line